### PR TITLE
Upgrade to Java 21

### LINKED ISSUE #1013: Migrate Web3Signer build requirements to Java 21
Migrate Web3Signer build requirements to Java 21.

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR Description
Upgrade build and runtime to Java 21. Following changes are also made to achieve this:
- Upgraded to Gradle 8.9
- Various plugin versions updated
- Various dependencies versions updated
  - Besu and Teku 
- Owasp suppressions updated
- Removed gatling performance test because it is not compatible with latest Gradle.

Logs after upgrading to Java 21:
```
Web3SignerApp | Version = web3signer/v24.2.0+20-gb2fe8d6/linux-aarch_64/-eclipseadoptium-openjdk64bitservervm-java-21
```

## Fixed Issue(s)

fixes #1013 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

## Testing

- [x] I thought about testing these changes in a realistic/non-local environment.

### CODE REVIEW

**Review by jframe — COMMENTED** [review]
We need give some notice about the breaking change of requiring Java 21 before merging this in. So for this upcoming release we can place a notice in our changelog and the announcement about the upcoming change and next monthly release change to using Java 21.

File: .circleci/config.yml
Comment (jframe) [inline .circleci/config.yml:226]: can we just remove this step instead of doing the echo?

File: acceptance-tests/build.gradle
Comment (jframe) [inline acceptance-tests/build.gradle:83]: remove commented line

File: acceptance-tests/build.gradle
Comment (jframe) [inline acceptance-tests/build.gradle:101]: Does the downloads directory get created if doesn't exist? And will work on Windows with the hardcoded `/` directory separator?

File: build.gradle
Comment (jframe) [inline build.gradle:34]: Do need to use a beta version of spotless? it's not in production code so it's probably ok.

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/LogErrorHandler.java
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/LogErrorHandler.java:34]: More descriptive error would be nicer here along line of invalid URI due to missing host or port.

File: slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbLocker.java
Comment (jframe) [inline slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbLocker.java:22]: This formatting is a bit nasty, I'm guessing this spotless enforcing this though?

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/LogErrorHandler.java
Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/LogErrorHandler.java:34]: sure. This NPE happens when invalid host header is involved from the client. Latest vertx has started throwing an NPE. I'll update the error description.

File: slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbLocker.java
Comment (usmansaleem) [reply slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbLocker.java:22]: Let me see if I am able to remove that `;`.

Comment (usmansaleem) [reply slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbLocker.java]: removed ;

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/LogErrorHandler.java
Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/LogErrorHandler.java]: refactored it a bit.

File: build.gradle
Comment (usmansaleem) [reply build.gradle:34]: I was double minded on this one, and finally added it to catch any surprising stuff in advance. Since it is build-only as you mentioned, we should be alright about it.

File: acceptance-tests/build.gradle
Comment (usmansaleem) [reply acceptance-tests/build.gradle:98]: This works on nix without any problems. Need to check our whole build on Windows environment.

File: acceptance-tests/build.gradle
Comment (usmansaleem) [reply acceptance-tests/build.gradle]: done.

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/signer/Signer.java
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/signer/Signer.java:211]: This could be simplified to:
```response.getBody().asString().lines()```

Removing the need for convertToLines

File: CHANGELOG.md
Comment (jframe) [inline CHANGELOG.md:13]: Also, we need to mention the changing of metrics to use openmetrics standard, users will need to update their dashboard

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/signer/Signer.java
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/signer/Signer.java:207]: nit: IMHO the previous name was clearer

File: build.gradle
Comment (jframe) [inline build.gradle:139]: remove commented line

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/LogErrorHandler.java
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/LogErrorHandler.java:49]: I think the method name isn't quite right as the error case it doesn't return a requestUri. Would be better to return an empty and just include that error message in the handle method.

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/signer/Signer.java
Comment (usmansaleem) [reply acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/signer/Signer.java]: nice suggestion, updated.

File: CHANGELOG.md
Comment (usmansaleem) [reply CHANGELOG.md:14]: Updated changelog, let me know if wording needs to be changed.

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/LogErrorHandler.java
Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/LogErrorHandler.java]: agreed. Updated the method to return Optional.

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/LogErrorHandler.java
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/LogErrorHandler.java:30]: Not going to block on this, but the variable name is a bit misleading as sometimes it is an error message sometimes it's the requestUri.

**Review by jframe — APPROVED** [review]

**Pull request merged**
